packages/test/test-end-to-end-tests/src/test/refreshSerializedStateManager.spec.ts

```diff
@@ -0,0 +1,505 @@
+				{ url },
+				pendingState.pendingState,
+			);
+			const dataStore2 = (await container2.getEntryPoint()) as ITestFluidObject;
+			const string2 = await dataStore2.getSharedObject<SharedString>(stringId);
+			await waitForContainerConnection(container2);
+			await provider.ensureSynchronized();
+			assert.strictEqual(string2.getText(), "abcde");
+
+			string2.insertText(string1.getLength(), "f");
+			const pendingState2 = await container2.closeAndGetPendingLocalState?.();
+			assert.ok(pendingState2);
+			const attributes2 = getAndAssertAttributes(pendingState2);
+			// the base snapshot was refreshed. sequenceNumber would be the same in case we haven't
+			assert(attributes.sequenceNumber < attributes2.sequenceNumber);
+
+			const container3: IContainerExperimental = await loader.resolve({ url }, pendingState2);
+			const dataStore3 = (await container3.getEntryPoint()) as ITestFluidObject;
+			const string3 = await dataStore3.getSharedObject<SharedString>(stringId);
+			await waitForContainerConnection(container3);
+			await provider.ensureSynchronized();
+			assert.strictEqual(string1.getText(), string3.getText());
+			assert.strictEqual(string3.getText(), "abcdef");
+		});
+
+		it("can load offline", async function () {
+			const pendingOps = await getPendingState(
+				provider,
+				async (s) => {},
+				async (s) => {
+					s.insertText(s.getLength(), ` a`);
+				},
+			);
+			const attributes = getAndAssertAttributes(pendingOps.parsed);
+			string1.insertText(string1.getLength(), " b");
+			await waitForSummary();
+
+			const container2 = await loadOffline(provider, { url }, pendingOps.pendingState);
+			const dataStore2 = (await container2.container.getEntryPoint()) as ITestFluidObject;
+			const string2 = await dataStore2.getSharedObject<SharedString>(stringId);
+
+			// pending changes should be applied
+			assert.strictEqual(string2.getText(), " a");
+
+			// make more changes while offline
+			string2.insertText(string2.getLength(), ` c`);
+
+			container2.connect();
+			await waitForContainerConnection(container2.container);
+			await provider.ensureSynchronized();
+			assert.strictEqual(string1.getText(), " a c b");
+			assert.strictEqual(string2.getText(), " a c b");
+			const pendingState2 = await container2.container.closeAndGetPendingLocalState?.();
+			assert.ok(pendingState2);
+			const attributes2 = getAndAssertAttributes(pendingState2);
+			// the base snapshot was refreshed. sequenceNumber would be the same in case we haven't
+			assert(attributes.sequenceNumber < attributes2.sequenceNumber);
+			const container3: IContainerExperimental = await loader.resolve({ url }, pendingState2);
+			const dataStore3 = (await container3.getEntryPoint()) as ITestFluidObject;
+			const string3 = await dataStore3.getSharedObject<SharedString>(stringId);
+			assert.strictEqual(string3.getText(), " a c b");
+		});
+
+		it("does not refresh on snapshot fetch failure", async () => {
```

Comment:
Currently the test throws the error but in reality it will retry. I  think the only thing I need to do is make the error retriable

Reply:
Also the title is wrong, it should refresh.